### ISSUE TITLE
Running Shadow task and mqtt Task Simultaneously giving LoadProhibition Errror.

### ISSUE THREAD

**Issue by Raghav3107** [aws/amazon-freertos#3123]
Hello,
I want to run the Shadow Task and MQTT Task simultaneously at the latest version. I know shadow uses the MQTT to work but  I want to publish and receive the message from a different topic instead of a shadow topic. Simply you can say I want to run **_mqtt_demo_mutual_auth.c_** and **_shadow_demo_main.c_** simultaneously.

For that, I m using the shadow_demo_main.c file for shadow, and for MQTT I tried to use the API of the **iot_mqtt_api.c** file and **iot_mqttt_agent.c** for MQTT. 
In MQTT, I created the MQTT agent and then tried to subscribed to Topic. But when the program calls the  **IotMqtt_TimedSubscribe()** API it starts giving **LoadProhibition** error.

**LOG::**
```
70 816 [main] 
71 816 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:448] 72 816 [main] Cleaned up outgoing publish packet with packet id 3.

73 817 [main] 
74 818 [main] [INFO] [MQTT] [core_mqtt.c:886] 75 818 [main] Packet received. ReceivedBytes=122.76 818 [main] 
77 818 [main] [INFO] [MQTT] [core_mqtt.c:1045] 78 818 [main] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.79 820 [main] 
80 820 [main] [INFO] [MQTT] [core_mqtt.c:1058] 81 820 [main] State record updated. New state=MQTTPubAckSend.82 820 [main] 
83 821 [main] [INFO] [ShadowDemo] [shadow_demo_main.c:618] 84 821 [main] pPublishInfo->pTopicName:$aws/things/rgv_newAWStest/shadow/delete/rejected.85 822 [main] 
86 822 [main] [INFO] [ShadowDemo] [shadow_demo_main.c:412] 87 822 [main] /delete/rejected json payload:{"code":404,"message":"No shadow exists with name: 'rgv_newAWStest'"}.88 823 [main] 
89 824 [main] [INFO] [ShadowDemo] [shadow_demo_main.c:446] 90 824 [main] Error code is: 404.91 824 [main] 
92 825 [main] [INFO] [ShadowDemo] [shadow_demo_main.c:456] 93 825 [main] Error code:404.94 825 [main] 
95 866 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:814] 96 866 [main] UNSUBSCRIBE sent topic $aws/things/rgv_newAWStest/shadow/delete/accepted to broker.

97 867 [main] 
98 869 [main] [INFO] [MQTT] [core_mqtt.c:886] 99 869 [main] Packet received. ReceivedBytes=2.100 869 [main] 
101 869 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:469] 102 869 [main] MQTT_PACKET_TYPE_UNSUBACK.

103 870 [main] 
104 930 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:814] 105 930 [main] UNSUBSCRIBE sent topic $aws/things/rgv_newAWStest/shadow/delete/rejected to broker.

106 930 [main] 
107 933 [main] [INFO] [MQTT] [core_mqtt.c:886] 108 933 [main] Packet received. ReceivedBytes=2.109 933 [main] 
110 933 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:469] 111 933 [main] MQTT_PACKET_TYPE_UNSUBACK.

112 934 [main] 
113 994 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:751] 114 994 [main] SUBSCRIBE topic $aws/things/rgv_newAWStest/shadow/update/delta to broker.

115 994 [main] 
116 999 [main] [INFO] [MQTT] [core_mqtt.c:886] 117 999 [main] Packet received. ReceivedBytes=3.118 999 [main] 
119 999 [main] [INFO] [MQTT] [mqtt_demo_helpers.c:463] 120 999 [main] MQTT_PACKET_TYPE_SUBACK.

121 1000 [main] 
========================== RUNNING-MQTT-TASK ============================ 
122 1060 [main] [INFO] [MQTT_MutualAuth_Demo] [mqtt_connection.c:51] 123 1060 [main] xMQTTClientCreate--------------------124 1060 [main] 
125 1060 [main] [INFO] [MQTT_MutualAuth_Demo] [mqtt_connection.c:62] 126 1060 [main] Attempting to create MQTT Agent.
127 1061 [main] 
128 1062 [main] [INFO] [MQTT_MutualAuth_Demo] [mqtt_connection.c:70] 129 1062 [main] Successfully created MQTT Agent.
130 1063 [main] 
131 1063 [MqttRecieveTask] [INFO] [MQTT_MutualAuth_Demo] [mqtt_connection.c:406] 132 1063 [MqttRecieveTask] MQTTReceiveTask-------------------------133 1064 [MqttRecieveTask] 
134 1160 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:864] 135 1160 [PublishTask] the published payload:{"state":{"desired":{"powerOn":5}},"clientToken":"001160"} 
 136 1160 [PublishTask] 
137 1161 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:892] 138 1161 [PublishTask] PUBLISH sent for topic $aws/things/rgv_newAWStest/shadow/update to broker with packet ID 7.

139 1162 [PublishTask] 
140 1167 [PublishTask] [INFO] [MQTT] [core_mqtt.c:886] 141 1167 [PublishTask] Packet received. ReceivedBytes=2.142 1167 [PublishTask] 
143 1167 [PublishTask] [INFO] [MQTT] [core_mqtt.c:1162] 144 1167 [PublishTask] Ack packet deserialized with result: MQTTSuccess.145 1167 [PublishTask] 
146 1168 [PublishTask] [INFO] [MQTT] [core_mqtt.c:1175] 147 1168 [PublishTask] State record updated. New state=MQTTPublishDone.148 1168 [PublishTask] 
149 1170 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:484] 150 1170 [PublishTask] PUBACK received for packet id 7.

151 1171 [PublishTask] 
152 1171 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:448] 153 1171 [PublishTask] Cleaned up outgoing publish packet with packet id 7.

154 1172 [PublishTask] 
155 1173 [PublishTask] [INFO] [MQTT] [core_mqtt.c:886] 156 1173 [PublishTask] Packet received. ReceivedBytes=182.157 1173 [PublishTask] 
158 1174 [PublishTask] [INFO] [MQTT] [core_mqtt.c:1045] 159 1174 [PublishTask] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.160 1175 [PublishTask] 
161 1175 [PublishTask] [INFO] [MQTT] [core_mqtt.c:1058] 162 1175 [PublishTask] State record updated. New state=MQTTPubAckSend.163 1175 [PublishTask] 
164 1176 [PublishTask] [INFO] [ShadowDemo] [shadow_demo_main.c:618] 165 1176 [PublishTask] pPublishInfo->pTopicName:$aws/things/rgv_newAWStest/shadow/update/delta.166 1178 [PublishTask] 
167 1178 [PublishTask] [INFO] [ShadowDemo] [shadow_demo_main.c:479] 168 1178 [PublishTask] /update/delta json payload:{"version":7211,"timestamp":1619438645,"state":{"powerOn":5},"metadata":{"powerOn":{"timestamp":1619438645}},"clientToken":"001160"}.169 1179 [PublishTask] 
170 1181 [PublishTask] [INFO] [ShadowDemo] [shadow_demo_main.c:520] 171 1181 [PublishTask] version: 7211172 1181 [PublishTask] 
173 1182 [PublishTask] [INFO] [ShadowDemo] [shadow_demo_main.c:530] 174 1182 [PublishTask] version:7211, ulCurrentVersion:0 
175 1183 [PublishTask] 
176 1183 [PublishTask] [INFO] [ShadowDemo] [shadow_demo_main.c:563] 177 1183 [PublishTask] The new power on state newState:5, ulCurrentPowerOnState:0 
178 1184 [PublishTask] 
179 1225 [PublishTask] [INFO] [ShadowDemo] [shadow_demo_main.c:677] 180 1225 [PublishTask] Report to the state change: 5181 1225 [PublishTask] 
182 1225 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:864] 183 1225 [PublishTask] the published payload:{"state":{"reported":{"powerOn":5}},"clientToken":"001225"} 
 184 1226 [PublishTask] 
185 1227 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:892] 186 1227 [PublishTask] PUBLISH sent for topic $aws/things/rgv_newAWStest/shadow/update to broker with packet ID 8.

187 1228 [PublishTask] 
188 1229 [PublishTask] [INFO] [MQTT] [core_mqtt.c:886] 189 1229 [PublishTask] Packet received. ReceivedBytes=2.190 1229 [PublishTask] 
191 1229 [PublishTask] [INFO] [MQTT] [core_mqtt.c:1162] 192 1230 [PublishTask] Ack packet deserialized with result: MQTTSuccess.193 1230 [PublishTask] 
194 1231 [PublishTask] [INFO] [MQTT] [core_mqtt.c:1175] 195 1231 [PublishTask] State record updated. New state=MQTTPublishDone.196 1231 [PublishTask] 
197 1232 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:484] 198 1232 [PublishTask] PUBACK received for packet id 8.

199 1233 [PublishTask] 
200 1233 [PublishTask] [INFO] [MQTT] [mqtt_demo_helpers.c:448] 201 1233 [PublishTask] Cleaned up outgoing publish packet with packet id 8.

202 1235 [PublishTask] 
203 3064 [MqttRecieveTask] [INFO] [MQTT_MutualAuth_Demo] [mqtt_connection.c:309] 204 3064 [MqttRecieveTask] _modifySubscriptions-----------------------205 3064 [MqttRecieveTask] 
206 3064 [MqttRecieveTask] [INFO] [MQTT_MutualAuth_Demo] [mqtt_connection.c:328] 207 3064 [MqttRecieveTask] IotMqtt_TimedSubscribe----------------------==============
208 3066 [MqttRecieveTask] 
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.
Core 0 register dump:
PC      : 0x400d93d3  PS      : 0x00060930  A0      : 0x800d9614  A1      : 0x3ffdd6c0  
0x400d93d3: _subscriptionCommon at /home/horsemann/Desktop/WorkSpace/aws_latest_repository/personal-box-freertos-demo/build/../components/mqtt/src/iot_mqtt_api.c:1374

A2      : 0x00000003  A3      : 0x00000000  A4      : 0x3ffdd760  A5      : 0x00000001  
A6      : 0x00000001  A7      : 0x00000000  A8      : 0x00000000  A9      : 0x3ffc3a80  
A10     : 0x3ffd8f18  A11     : 0x00000001  A12     : 0x3ffdd760  A13     : 0x00000001  
A14     : 0x00000000  A15     : 0x3ffd8f20  SAR     : 0x00000013  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x400014fd  LEND    : 0x4000150d  LCOUNT  : 0xffffffff  

ELF file SHA256: 70b5cdcb110c1f5c

Backtrace: 0x400d93d3:0x3ffdd6c0 0x400d9611:0x3ffdd700 0x400d972c:0x3ffdd730 0x400d1c0b:0x3ffdd760 0x400d1d45:0x3ffdd7a0
0x400d93d3: _subscriptionCommon at /home/horsemann/Desktop/WorkSpace/aws_latest_repository/personal-box-freertos-demo/build/../components/mqtt/src/iot_mqtt_api.c:1374

0x400d9611: IotMqtt_Subscribe at /home/horsemann/Desktop/WorkSpace/aws_latest_repository/personal-box-freertos-demo/build/../components/mqtt/src/iot_mqtt_api.c:1374

0x400d972c: IotMqtt_TimedSubscribe at /home/horsemann/Desktop/WorkSpace/aws_latest_repository/personal-box-freertos-demo/build/../components/mqtt/src/iot_mqtt_api.c:1515

0x400d1c0b: _modifySubscriptions at /home/horsemann/Desktop/WorkSpace/aws_latest_repository/personal-box-freertos-demo/build/../src/mqtt_connection.c:329 (discriminator 4)

0x400d1d45: MqttRecieveTask at /home/horsemann/Desktop/WorkSpace/aws_latest_repository/personal-box-freertos-demo/build/../src/mqtt_connection.c:420
```
**CODE:**
```
int _modifySubscriptions(IotMqttOperationType_t operation,
                         const char **pTopicFilters,
                         void *pCallbackParameter)
{
    LogInfo(("_modifySubscriptions-----------------------"));
    int status = EXIT_SUCCESS;
    int32_t i = 0;
    IotMqttError_t subscriptionStatus = IOT_MQTT_STATUS_PENDING;
    IotMqttSubscription_t pSubscriptions[TOPIC_FILTER_COUNT] = {IOT_MQTT_SUBSCRIPTION_INITIALIZER};

    /* Set the members of the subscription list. */
    for (i = 0; i < TOPIC_FILTER_COUNT; i++)
    {
        pSubscriptions[i].qos = IOT_MQTT_QOS_1;
        pSubscriptions[i].pTopicFilter = pTopicFilters[i];
        pSubscriptions[i].topicFilterLength = strlen(pTopicFilters[i]);
        pSubscriptions[i].callback.pCallbackContext = pCallbackParameter;
        pSubscriptions[i].callback.function = _mqttSubscriptionCallback;
    }

    /* Modify subscriptions by either subscribing or unsubscribing. */
    if (operation == IOT_MQTT_SUBSCRIBE)
    {
        LogInfo(("IotMqtt_TimedSubscribe----------------------==============\n"));
        subscriptionStatus = IotMqtt_TimedSubscribe(mqttConnection,
                                                    pSubscriptions,
                                                    TOPIC_FILTER_COUNT,
                                                    0,
                                                    MQTT_TIMEOUT_MS);

        /* Check the status of SUBSCRIBE. */
        switch (subscriptionStatus)
        {
        case IOT_MQTT_SUCCESS:
            LogInfo(("All demo topic filter subscriptions accepted."));
            break;

        case IOT_MQTT_SERVER_REFUSED:

            /* Check which subscriptions were rejected before exiting the demo. */
            for (i = 0; i < TOPIC_FILTER_COUNT; i++)
            {
                if (IotMqtt_IsSubscribed(mqttConnection,
                                         pSubscriptions[i].pTopicFilter,
                                         pSubscriptions[i].topicFilterLength,
                                         NULL) == true)
                {
                    LogInfo(("Topic filter %.*s was accepted.",
                             pSubscriptions[i].topicFilterLength,
                             pSubscriptions[i].pTopicFilter));
                }
                else
                {
                    LogError(("Topic filter %.*s was rejected.",
                              pSubscriptions[i].topicFilterLength,
                              pSubscriptions[i].pTopicFilter));
                }
            }

            status = EXIT_FAILURE;
            break;

        default:

            status = EXIT_FAILURE;
            break;
        }
    }
    else if (operation == IOT_MQTT_UNSUBSCRIBE)
    {
        subscriptionStatus = IotMqtt_TimedUnsubscribe(mqttConnection,
                                                      pSubscriptions,
                                                      TOPIC_FILTER_COUNT,
                                                      0,
                                                      MQTT_TIMEOUT_MS);

        /* Check the status of UNSUBSCRIBE. */
        if (subscriptionStatus != IOT_MQTT_SUCCESS)
        {
            status = EXIT_FAILURE;
        }
    }
    else
    {
        /* Only SUBSCRIBE and UNSUBSCRIBE are valid for modifying subscriptions. */
        LogError(("MQTT operation %s is not valid for modifying subscriptions.",
                  IotMqtt_OperationType(operation)));

        status = EXIT_FAILURE;
    }

    return status;
}

/***************************************************/
/**
 * @brief : Used to receive mqtt message from suscribe topic
 * 
 */
void MqttRecieveTask()
{
    LogInfo(("MQTTReceiveTask-------------------------"));
    vTaskDelay(2000);
    /* Counts the number of incoming PUBLISHES received (and allows the demo
   * application to wait on incoming PUBLISH messages). */
    IotSemaphore_t publishesReceived;

    /* Topics used as both topic filters and topic names in this demo. */
    const char *pTopics[TOPIC_FILTER_COUNT] =
        {
            IOT_DEMO_MQTT_TOPIC_PREFIX "/topic/test",
            IOT_DEMO_MQTT_TOPIC_PREFIX "/topic/2",
            IOT_DEMO_MQTT_TOPIC_PREFIX "/topic/3",
            IOT_DEMO_MQTT_TOPIC_PREFIX "/devices/test",
        };
    int status = _modifySubscriptions(IOT_MQTT_SUBSCRIBE,
                                      pTopics,
                                      &publishesReceived);
    while (1)
    {

        vTaskDelay(100);
    }
}

void runMqttTask(void)
{
    printf("========================== RUNNING-MQTT-TASK ============================ \n");
    xMQTTClientCreate();

    static TaskHandle_t hMQTTReceive = NULL;
    xTaskCreate(MqttRecieveTask, "MqttRecieveTask", 4096, NULL, 2, &hMQTTReceive);
    return;
}
```
Thanks in advance for your help.

**Comment by paulbartell** [comment]
@Raghav3107 Using mqtt from multiple tasks requires either multiple mqtt connections or the use of another task to route messages. 

Our coreMQTT-Agent project is meant to address this need, but has not exited FreeRTOS Labs yet. Please check out the [CoreMQTT-Agent repository](https://github.com/FreeRTOS/coreMQTT-Agent) and [CoreMQTT-Agent-Demo](https://github.com/FreeRTOS/coreMQTT-Agent-Demos). Feel free to re-open this issue or inquire further on [FreeRTOS Forums](https://forums.freertos.org/).
